### PR TITLE
Use memcache versioning for autotune remote cache

### PR DESCRIPTION
**Pull request by oulgen**
Summary: Internal training platform doesn't get updated very frequently, so lets use versioning for memcache.

Test Plan: existing tests

Differential Revision: D54818197

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by aakhundov — APPROVED** [review]

File: torch/_utils_internal.py [hunk 1]
```diff
@@ -121,6 +121,13 @@ def justknobs_check(name: str) -> bool:
     return True
 
 
+def justknobs_getval_int(name: str) -> int:
+    """
+    Read warning on justknobs_check
+    """
+    return 0
```
Comment (jansel) [inline torch/_utils_internal.py:128]: Are we doing some weird monkeypatching in fbcode?  Let's not do that, it makes things hard to follow.  You can create a `/fb/` folder that won't get exported to github.

**Review by jansel — CHANGES_REQUESTED** [review]

File: torch/_utils_internal.py (hunk 1 above)
Comment (oulgen) [reply torch/_utils_internal.py:128]: I did not add any monkey patching, there's an fb version of this file. This file contains the stubs. Please look at D54818197

Comment (oulgen) [reply torch/_utils_internal.py:128]: I followed @ezyang 's example from https://github.com/pytorch/pytorch/pull/120915

Comment (bertmaher) [reply torch/_utils_internal.py:128]: We selectively include fb/_utils_internal.py in fbcode vs torch/_utils_internal.py in open source.  It's not monkeypatching exactly but it is a bit unusual.  I think @oulgen is just following the pattern for JK killswitches introduced by @ezyang above.

Comment (bertmaher) [reply torch/_utils_internal.py:128]: (Oh ha you said exactly the same thing in the span of time it took me to write my comment :-p )

**Review by jansel — APPROVED** [review]
